spec/ameba/rule/typing/method_return_type_restriction_spec.cr

```diff
@@ -0,0 +1,117 @@
+require "../../../spec_helper"
+
+module Ameba::Rule::Typing
+  subject = MethodReturnTypeRestriction.new
+  subject.private_methods = true
+  subject.protected_methods = true
+  subject.undocumented = true
```

Comment:
at top level we usually test the default config, then the props can be toggled in a separate context.